net/Socket.cpp

```diff
@@ -555,6 +555,33 @@ void StreamSocket::dumpState(std::ostream& os)
         Util::dumpHex(os, "\t\toutBuffer:\n", "\t\t", _outBuffer);
 }
 
+void StreamSocket::sendError(int errorCode,
```

Comment:
In terms of interface, the shutdown logic doesn't seem to belong to a "send error" function.

I'd suggest splitting it into an explicit sendError and sendErrorAndShutdown functions.

Internally, I'd have the sendErrorAndShutdown call sendError and then call the shutdown() function.

In addition, the body and extraHeader arguments can either be used in an overloaded version of sendError, or defaulted (to std::string(), not to ""). That way, we don't need to see all the "" arguments when calling sendError.

Reply:
resolved in 7fa7ee5e3